src/components/SearchField.tsx

```diff
@@ -1,38 +1,241 @@
+        );
+      }
+    },
+    [inputValue, history, checkboxes, genreToSearch, dateRange]
+  );
+
+  const onFocusSearchInput = useCallback(
+    (event: React.SyntheticEvent): void => {
+      setFocus(true);
+    },
+    []
+  );
+
+  const onBlurSearchInput = useCallback((event: React.SyntheticEvent): void => {
+    setFocus(false);
+  }, []);
+
+  const mouseEnterDropdown = useCallback(
+    (event: React.SyntheticEvent): void => {
+      setDropdown(true);
+    },
+    []
+  );
+
+  const mouseLeaveDropdown = useCallback(
+    (event: React.SyntheticEvent): void => {
+      setDropdown(false);
+    },
+    []
+  );
+
+  const fromDateInputIncrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.fromDate >= dateRange.toDate) return;
+      setDateRange({ ...dateRange, fromDate: dateRange.fromDate + 1 });
+    },
+    [dateRange]
+  );
+
+  const fromDateInputDecrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.fromDate <= 1980) return;
+      setDateRange({ ...dateRange, fromDate: dateRange.fromDate - 1 });
+    },
+    [dateRange]
+  );
+
+  const toDateInputIncrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.toDate >= 2022) return;
+      setDateRange({ ...dateRange, toDate: dateRange.toDate + 1 });
+    },
+    [dateRange]
+  );
+
+  const toDateInputDecrease = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+      if (dateRange.toDate <= dateRange.fromDate) return;
+      setDateRange({ ...dateRange, toDate: dateRange.toDate - 1 });
+    },
+    [dateRange]
+  );
+
+  const favoritesCheckboxHandler = useCallback(
+    (event: React.InputHTMLAttributes<HTMLInputElement>): void => {
+      setCheckboxes({ ...checkboxes, favorites: !checkboxes.favorites });
+    },
+    [checkboxes]
+  );
+
+  const historyCheckboxHandler = useCallback(
+    (event: React.InputHTMLAttributes<HTMLInputElement>): void => {
+      setCheckboxes({ ...checkboxes, history: !checkboxes.history });
+    },
+    [checkboxes]
+  );
+
+  const selectGenreHandler = useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>): void => {
+      setGenreToSearch(event.target.value);
+    },
+    []
+  );
 
   return (
     <>
-      <form onSubmit={submitHandler} className="search-form">
+      <form onSubmit={submitSearchRequest} className="search-form">
         <input
+          onBlur={onBlurSearchInput}
           value={inputValue}
-          onChange={changeHandler}
+          onChange={searchInputChange}
           id="search"
           type="search"
           placeholder="Search"
           className="header-search"
+          onFocus={onFocusSearchInput}
```

Comment:
Prefix "on" usually means event, the name of function is a verb

Reply:
replaced it with "whenFocusSearchInput"